### ISSUE TITLE
Confusion matrix 

### ISSUE THREAD

**Issue by mezny** [mezny/Convnet#1]
Hello ``
I am trying to detect emotion from the speech. for this I must Use the confusion matrix from Tnesorflow , I could not know how to use it .. I must detect the 7 emotions... 
my result for training accuracy is 67% and tetsing is 100 % ( I know it can be not true ) but this what I got . 
Now about the confusion Matrix 
# Here what I wrote : 

confusion_matrix_tf = tf.confusion_matrix(tf.argmax(Y_, 1), tf.argmax(Y, 1), num_classes=7, dtype=tf.int32, name=None, weights=None)

with tf.Session() as session:
    tf.global_variables_initializer().run()
    for epoch in range(training_epochs):
        cost_history = np.empty(shape=[1],dtype=float)
        for b in range(total_batches):    
            offset = (b * batch_size) % (train_Y.shape[0] - batch_size)
            batch_X = train_X[offset:(offset + batch_size), :, :, :]
            batch_Y = train_Y[offset:(offset + batch_size), :]
            _, c = session.run([optimizer, cross_entropy],feed_dict={X: batch_X, Y : batch_Y})
            cost_history = np.append(cost_history,c)
        print( "Epoch: ",epoch," Training Loss: ",np.mean(cost_history)," Training Accuracy: ", session.run(accuracy, feed_dict={X: train_X, Y: train_Y, keep_prob : dropout_}))
        
    print ("Testing Accuracy:", session.run(accuracy, feed_dict={X: test_X, Y: test_Y,keep_prob : 1.0}), session.run)
    print("Valid data accuracy:", accuracy.eval({X: valid_X, Y: valid_Y, keep_prob : 1.0})) 
    print ("c_m : " , session.run(confusion_matrix_tf, feed_dict={X: test_X, Y: test_Y, keep_prob: 1.0})) 
    

HOW DO I FIX MY COnfusion matrix , it gave me like a result 

c_m :  [[0 0 0 ..., 0 0 0]
 [0 0 0 ..., 0 0 0]
 [3 0 0 ..., 0 0 0]
 ..., 
 [0 0 0 ..., 0 0 0]
 [0 0 0 ..., 0 0 0]
 [0 0 0 ..., 0 0 0]]
